test/e2e/kubevirt.go

```diff
@@ -2359,4 +2360,97 @@ chpasswd: { expire: False }
 		)
 	})
 
+	Context("MAC conflicts detection", func() {
+		newTestVMI := func(name, mac string) *kubevirtv1.VirtualMachineInstance {
+			vm := fedoraWithTestToolingVMI(nil, nil, nil, kubevirtv1.NetworkSource{Pod: &kubevirtv1.PodNetwork{}}, "#", "")
+			vm.Name = name
+			vm.Spec.Domain.Devices.Interfaces[0].Bridge = nil
+			vm.Spec.Domain.Devices.Interfaces[0].Binding = &kubevirtv1.PluginBinding{Name: "l2bridge"}
+			vm.Spec.Domain.Devices.Interfaces[0].MacAddress = mac
+			return vm
+		}
+
+		It("should fail when creating second VM with duplicate static IP", func() {
+			if !isPreConfiguredUdnAddressesEnabled() {
+				Skip("ENABLE_PRE_CONF_UDN_ADDR not configured")
+			}
+			By("Create test namespace")
+			l := map[string]string{
+				"e2e-framework":           fr.BaseName,
+				RequiredUDNNamespaceLabel: "",
+			}
+			ns, err := fr.CreateNamespace(context.TODO(), fr.BaseName, l)
+			Expect(err).NotTo(HaveOccurred())
+			fr.Namespace = ns
+			namespace = fr.Namespace.Name
+
+			cudnName := uniqueMetaName("test")
+			dualCIDRs := filterDualStackCIDRs(fr.ClientSet, []udnv1.CIDR{udnv1.CIDR("10.128.0.0/24"), udnv1.CIDR("2010:100:200::0/60")})
+			testCudn, _ := kubevirt.GenerateCUDN(namespace, cudnName, udnv1.NetworkTopologyLayer2, udnv1.NetworkRolePrimary, dualCIDRs)
+			createCUDN(testCudn)
+
+			testMAC := "02:a1:b2:c3:d4:e5"
+			vmi1 := newTestVMI("vm-static-mac", testMAC)
+			vm1 := generateVM(vmi1)
+			vm1.Name = vmi1.Name
+			createVirtualMachine(vm1)
+
+			By("Asserting VM with static MAC is running as expected")
+			Eventually(func(g Gomega) []kubevirtv1.VirtualMachineInstanceCondition {
+				err := crClient.Get(context.Background(), crclient.ObjectKeyFromObject(vmi1), vmi1)
+				g.Expect(err).To(SatisfyAny(WithTransform(apierrors.IsNotFound, BeTrue()), Succeed()))
```

Comment:
why do you have the `SatisfyAny` matcher here ? you're only applying one transform to the input data.